include/transformer.h

```diff
@@ -82,6 +82,19 @@ class LargeSquare64x64Transformer : public CanvasTransformer {
   TransformCanvas *const canvas_;
 };
 
+class HalfTransformer : public CanvasTransformer {
+public:
+  HalfTransformer();
```

Comment:
This now only works for 32->16 pixel mapping (see below my SetPixel() comment).

Since the transformer per-se does not know if it is called in a 32 or 16 context (consider a 8 pixel high display, that is behaving like 1/2 a 16 display), this should probably have a constructor parameter.

HalfTransformer::HalfTransformer(int half_height) ?

Then, when you get a delegatee, you can assert(canvas->height() % (2*half_height) == 0);